smf/src/lib.rs

```diff
@@ -172,6 +175,51 @@ impl Scf {
     pub fn scopes(&self) -> Result<Scopes> {
         Scopes::new(self)
     }
+
+    pub fn get_instance_from_fmri(&self, fmri: &str) -> Result<Instance<'_>> {
+        let fmri = CString::new(fmri).unwrap();
+
+        let scope = Scope::new(self)?;
+        let service = Service::new(self)?;
+        let instance = Instance::new(self)?;
+
+        let ret = unsafe {
+            scf_handle_decode_fmri(
+                self.handle.as_ptr(),
+                fmri.as_ptr(),
+                scope.scope.as_ptr(),
+                service.service.as_ptr(),
+                instance.instance.as_ptr(),
+                ptr::null_mut(),
+                ptr::null_mut(),
+                // `fmri` _must_ have an instance, but is allowed to also
+                // specify a propertygroup / property; ignore those if present
+                // via `SCF_DECODE_FMRI_TRUNCATE`.
+                SCF_DECODE_FMRI_REQUIRE_INSTANCE | SCF_DECODE_FMRI_TRUNCATE,
+            )
+        };
+
+        if ret == 0 {
+            Ok(instance)
+        } else {
+            Err(ScfError::last())
+        }
+    }
+
+    /// From within a running service instance, get our own [`Instance`].
+    ///
+    /// If you are using this to look up the current value of your properties,
+    /// you almost certainly want to call [`Instance::get_running_snapshot()`]
+    /// on the returned instance.
+    ///
+    /// This method looks up our own FMRI via the `SMF_FMRI` environment
+    /// variable, which is supplied by `smf` to running instances.
```

Comment:
I would write **SMF** rather than `smf` in prose -- I'm sure I've (and the manual has) been sloppy here, but it's an acronym, etc.  C-style comments, etc.